chain/client/src/client_actor.rs

```diff
@@ -1774,6 +1783,37 @@ impl ClientActor {
         }
     }
 
+    /// Verifies if the node possesses sync block.
+    /// It is the last block of the previous epoch.
+    /// If the block is absent, the node requests it from peers.
+    fn sync_block_status(
+        &self,
+        prev_hash: &CryptoHash,
+        now: DateTime<Utc>,
+    ) -> Result<(bool, bool), near_chain::Error> {
+        let (request_block, have_block) = if !self.client.chain.block_exists(prev_hash)? {
+            let timeout = self.client.config.state_sync_timeout.as_millis() as i64;
```

Comment:
I know I sound like a broken record.. but I don't see the point of this `as_millis()` either really. Of course it's not as much of a problem as the `as_secs()`, but even just as a matter of style it seems strange to me. Why not just compare the values?

```diff
diff --git a/chain/client/src/client_actor.rs b/chain/client/src/client_actor.rs
index 64789f378..08cf5bdaa 100644
--- a/chain/client/src/client_actor.rs
+++ b/chain/client/src/client_actor.rs
@@ -1792,15 +1792,15 @@ impl ClientActor {
         now: DateTime<Utc>,
     ) -> Result<(bool, bool), near_chain::Error> {
         let (request_block, have_block) = if !self.client.chain.block_exists(prev_hash)? {
-            let timeout = self.client.config.state_sync_timeout.as_millis() as i64;
+            let timeout = chrono::Duration::from_std(self.client.config.state_sync_timeout).unwrap();
             match self.client.last_time_sync_block_requested {
                 None => (true, false),
                 Some(last_time) => {
-                    if (now - last_time).num_milliseconds() >= timeout {
+                    if (now - last_time) >= timeout {
                         tracing::error!(
                             target: "sync",
                             %prev_hash,
-                            timeout_sec = timeout as f64 * 1e-3,
+                            ?timeout,
                             "State sync: block request timed out");
                         (true, false)
                     } else {
```

the `unwrap()` will not fail because the timeout value in the config is not gonna be above `chrono::duration::MAX`